torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -27,6 +27,8 @@ std::shared_ptr<Operator> matchBuiltinOp(
     const py::args& args,
     const py::kwargs& kwargs,
     Stack& stack) {
+  // Acquire GIL for py::args and py::kwargs processing.
+  pybind11::gil_scoped_acquire ag;
   Symbol symbol = Symbol::fromQualString(opName);
   if (symbol.is_aten()) {
```

Comment:
Since `matchBuiltinOp(..)` is called in both `pyRpcBuiltin` and `pyRemoteBuiltin`, will this incur Python thread context switch cost for these two entries? 
It's releases GIL on pybind11 call, but after pybind11 call dispatch, it immediately acquires GIL.

Is it better if we release GIL after the call to `matchBuiltinOp(..)`?

```cpp
  module.def(
      "_invoke_rpc_builtin",
      [](const WorkerInfo& dst,
         const std::string& opName,
         const std::shared_ptr<torch::autograd::profiler::RecordFunction>& rf,
         const py::args& args,
         const py::kwargs& kwargs) {
        return pyRpcBuiltin(dst, opName, rf, args, kwargs);
      }, py::call_guard<gil_scoped_acquire>);
```

```cpp
std::shared_ptr<FutureMessage> pyRpcBuiltin(
    const WorkerInfo& dst,
    const std::string& opName,
    const std::shared_ptr<torch::autograd::profiler::RecordFunction>& rf,
    const py::args& args,
    const py::kwargs& kwargs) {
  Stack stack;
  auto op = matchBuiltinOp(opName, args, kwargs, stack);
  py::gil_scoped_release release;
  auto scriptCall = std::make_unique<ScriptCall>(op, std::move(stack));
}
```

Reply:
Good point, I'll reorder this.